### PR TITLE
[API][News] Add get news endpoint

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Api/Filter/JecouteNewsScopeFilter.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+use Symfony\Component\Security\Core\Security;
+
+final class JecouteNewsScopeFilter extends AbstractContextAwareFilter
+{
+    private const PROPERTY_NAME = 'scope';
+    private const OPERATION_NAMES = ['get_private'];
+
+    private GeneralScopeGenerator $generalScopeGenerator;
+    private Security $security;
+    private AdherentMessageRepository $adherentMessageRepository;
```
Comment (ottaviano) [inline src/Api/Filter/JecouteNewsScopeFilter.php:23]: no needed here

File: src/Api/Serializer/JecouteNewsGroupsContextBuilder.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace App\Api\Serializer;
+
+use App\Entity\Jecoute\News;
+
+class JecouteNewsGroupsContextBuilder extends AbstractGroupsContextBuilder
```
Comment (ottaviano) [inline src/Api/Serializer/JecouteNewsGroupsContextBuilder.php:7]: why do we need this class?

File: src/Normalizer/JecouteNewsNormalizer.php [hunk 3]
```diff
@@ -37,7 +37,8 @@ public function supportsNormalization($data, $format = null, array $context = []
         return
             empty($context[self::NEWS_ALREADY_CALLED])
             && $data instanceof News
-            && \in_array('jecoute_news_read', $context['groups'] ?? [])
+            && (\in_array('jecoute_news_read', $context['groups'] ?? []) ||
```
Comment (ottaviano) [inline src/Normalizer/JecouteNewsNormalizer.php:40]: what about using `array_intersect` instead?

File: src/Api/Serializer/JecouteNewsGroupsContextBuilder.php (hunk 2 above)
Comment (elvino29) [reply src/Api/Serializer/JecouteNewsGroupsContextBuilder.php:7]: I think that with the separation of endpoints, we no longer need this context builder

File: src/Normalizer/JecouteNewsNormalizer.php (hunk 3 above)
Comment (elvino29) [reply src/Normalizer/JecouteNewsNormalizer.php:40]: Like `count(array_intersect($array1, $array2)) !== 0` ?

File: src/Entity/Jecoute/News.php [hunk 4]
```diff
@@ -45,12 +52,19 @@
...
+ *         "get_private": {
+ *             "path": "/v3/jecoute/news",
+ *             "method": "GET",
+ *             "normalization_context": {"groups": {"jecoute_news_read_dc"}},
+ *             "access_control": "is_granted('IS_FEATURE_GRANTED', 'news')",
+ *         },
  *     },
  *     itemOperations={
- *         "get": {
+ *         "get_public": {
```
Comment (ottaviano) [inline src/Entity/Jecoute/News.php:64]: could you add the security check for this route, please? 
`"access_control": "is_granted('ROLE_OAUTH_SCOPE_JEMARCHE_APP')",`

**Review by ottaviano — APPROVED** [review]

**Pull request merged**